packages/backend/src/open_payments/receiver/service.ts

```diff
@@ -45,10 +61,74 @@ export async function createReceiverService(
   }
 
   return {
-    get: (url) => getReceiver(deps, url)
+    get: (url) => getReceiver(deps, url),
+    create: (url) => createReceiver(deps, url)
   }
 }
 
+async function createReceiver(
+  deps: ServiceDependencies,
+  args: CreateReceiverArgs
+): Promise<Receiver> {
+  const localPaymentPointer = await deps.paymentPointerService.getByUrl(
+    args.paymentPointerUrl
+  )
+
+  const incomingPayment = localPaymentPointer
+    ? await createLocalIncomingPayment(deps, args, localPaymentPointer)
+    : await deps.remoteIncomingPaymentService.create(args)
+
+  const receiver = Receiver.fromIncomingPayment(incomingPayment)
+
+  if (!receiver) {
+    const errorMessage = 'Could not create receiver from incoming payment'
+    deps.logger.error({ incomingPayment }, errorMessage)
+
+    throw new Error(errorMessage)
+  }
+
+  return receiver
+}
+
+async function createLocalIncomingPayment(
+  deps: ServiceDependencies,
+  args: CreateReceiverArgs,
+  paymentPointer: PaymentPointer
+): Promise<OpenPaymentsIncomingPayment> {
+  const { description, expiresAt, incomingAmount, externalRef } = args
+
+  const incomingPaymentOrError = await deps.incomingPaymentService.create({
+    paymentPointerId: paymentPointer.id,
+    description,
+    expiresAt,
+    incomingAmount,
+    externalRef
+  })
+
+  if (isIncomingPaymentError(incomingPaymentOrError)) {
+    const errorMessage = 'Could not create local incoming payment'
+    deps.logger.error(
+      { error: incomingPaymentErrorMessageMap[incomingPaymentOrError] },
+      errorMessage
+    )
+
+    throw new Error(errorMessage)
+  }
+
+  const connection = deps.connectionService.get(incomingPaymentOrError)
+
+  if (!connection) {
+    const errorMessage = 'Could not get connection for local incoming payment'
+    deps.logger.error({ incomingPaymentOrError }, errorMessage)
+
+    throw new Error(errorMessage)
+  }
+
+  return incomingPaymentOrError.toOpenPaymentsType({
+    ilpStreamConnection: connection
+  })
```

Comment:
Once I split out the `IncomingPayment` types in `open-payments` with and without the ilpStreamConnection
- https://github.com/interledger/rafiki/issues/662

The corresponding connection fetching will (should?) be optional